io.openems.edge.controller.ess.limitusablecapacity/src/io/openems/edge/controller/ess/limitusablecapacity/LimitUsableCapacityControllerImpl.java

```diff
@@ -0,0 +1,181 @@
+package io.openems.edge.controller.ess.limitusablecapacity;
+
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.exceptions.OpenemsException;
+import io.openems.edge.common.channel.value.Value;
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.ComponentManager;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.controller.api.Controller;
+import io.openems.edge.ess.api.ManagedSymmetricEss;
+import io.openems.edge.ess.power.api.Constraint;
+import io.openems.edge.ess.power.api.Phase;
+import io.openems.edge.ess.power.api.Pwr;
+import io.openems.edge.ess.power.api.Relationship;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Controller.Ess.LimitUsableCapacity", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE //
+)
+public class LimitUsableCapacityControllerImpl extends AbstractOpenemsComponent
+		implements LimitUsableCapacityController, Controller, OpenemsComponent {
+
+	@Reference
+	protected ComponentManager componentManager;
+
+	// Force charge power
+	private int forceChargePower = 2000;
+
+	private Config config;
+	private State state = State.UNDEFINED;
+
+	public LimitUsableCapacityControllerImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				Controller.ChannelId.values(), //
+				LimitUsableCapacityController.ChannelId.values() //
+		);
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) throws OpenemsException {
+		super.activate(context, config.id(), config.alias(), config.enabled());
+
+		this.config = config;
+
+		/*
+		 * Checking the Soc values in the configuration
+		 * 
+		 * forceChargeSoc < stopDischargeSoc < allowDischargeSoc < allowChargeSoc <
+		 * stopChargeSoc
+		 * 
+		 */
+		if (this.config.forceChargeSoc() > this.config.stopDischargeSoc()
+				&& this.config.stopDischargeSoc() > this.config.allowDischargeSoc()
+				&& this.config.allowDischargeSoc() > this.config.allowChargeSoc()
+				&& this.config.allowChargeSoc() > this.config.stopChargeSoc()) {
+			throw new OpenemsException("Please re-check the configuration, invalid values present in the Soc columns");
+		}
+	}
+
+	@Override
+	public void run() throws OpenemsNamedException {
+		ManagedSymmetricEss ess = this.componentManager.getComponent(this.config.ess_id());
+
+		// Set to normal state and return if SoC is not available
+		Value<Integer> socOpt = ess.getSoc();
+		if (!socOpt.isDefined()) {
+			this.state = State.NO_LIMIT;
+			return;
```

Comment:
If you return here, your Channels are never written, i.e. the `STATE_MACHINE` Channel can never be `NO_LIMIT` and also if the SoC of a ESS disappears (e.g. cable was removed), `ALLOWED_CHARGE_POWER` and `ALLOWED_DISCHARGE_POWER` will still show the old values; even though the limit is not applied.

Reply:
Updated this.